dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -78,4 +81,44 @@ public static String formatProblems(Iterable<ArtifactProblem> problems) {
     }
     return output.toString();
   }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other == null || getClass() != other.getClass()) {
+      return false;
+    }
+
+    ArtifactProblem otherProblem = (ArtifactProblem) other;
+    return Objects.equals(artifact, otherProblem.artifact)
+        && sameDependencyPath(dependencyPath, otherProblem.dependencyPath);
+  }
+
+  private static boolean sameDependencyPath(
+      ImmutableList<DependencyNode> listA, ImmutableList<DependencyNode> listB) {
+    int size = listA.size();
+    if (listB.size() != size) {
+      return false;
+    }
+
+    for (int i = 0; i < size; i++) {
+      DependencyNode nodeA = listA.get(i);
+      DependencyNode nodeB = listB.get(i);
+      Dependency dependencyA = nodeA.getDependency();
+      Dependency dependencyB = nodeB.getDependency();
+      if (!Objects.equals(dependencyA, dependencyB)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    ImmutableList<Dependency> dependencyList =
```

Comment:
DependencyNode does not have good equals/hashcode. So leveraging Dependency class.